api/output_string.go

```diff
@@ -46,6 +48,22 @@ func (d *OutputStringError) parseRequest() {
 	if d.Request.Method != "GET" {
 		d.parsedCurlString = fmt.Sprintf("%s-X %s ", d.parsedCurlString, d.Request.Method)
 	}
+	if d.ClientCACert != "" {
+		clientCACert := strings.Replace(d.ClientCACert, "'", "'\"'\"'", -1)
```

Comment:
```suggestion
		clientCACert := strings.Replace(d.ClientCACert, "'", `'"'"'`, -1)
```

I'm still not sure I understand it even with that change.

Reply:
This is so that if the filename contains a special character:

```
$ ls
clie'nt.crt client.key
```
then the `'` is escaped in the output of `-output-curl-string`:
```
$ ../bin/vault read -client-cert clie\'nt.crt -client-key client.key -output-curl-string foo
curl --cert 'clie'"'"'nt.crt' --key 'client.key' -H "X-Vault-Request: true" -H "X-Vault-Token: $(vault print token)" https://127.0.0.1:8200/v1/foo
```

The `--cert` option is the concatenation of three strings `'clie'`, `"'"` and `nt.crt` so it gives `"clie'nt.crt"`.

For what it's worth this is very naive, it does not escape `$(...)` or ` `` ` which would also interpreted by the shell. I don't think this is a solvable problem thought as different shells will interpret different constructs. I used this method to escape `'` because it is already used for the body.